src/shared/file-descriptor.h

```diff
@@ -0,0 +1,30 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+#pragma once
+
+#include "list.h"
+#include "macro.h"
+
+typedef struct FileDescriptor {
```

Comment:
The name is a bit too generic, I'd go for `ExtraFileDescriptor`

Reply:
Also, I think we can keep this in src/core since it's not used anywhere outside of it.

Reply:
Cool I'll put it in src/core/extra-file-descriptor.c/h!